modules/compute/function-app/main.bicep

```diff
@@ -0,0 +1,299 @@
+/*param section*/
+@description('Required. Name for the Azure Function App.')
+param name string
```

Comment:
```suggestion
@minLength(6)
@maxLength(64)
param name string = take('func${uniqueString(resourceGroup().id, subscription().id, location)}', 64)
```

Reply:
Tests are currently failing with `Status Message: The provided deployment name 'func-app2-267-4b15b98-20230408010517-f4e616e5fc1b4c2d813dddebe66c6d6a' has a length of '69' which exceeds the maximum length of '64'.`

This will help catch that earlier.

Reply:
I think adding `@maxLength(64)` makes sense. We should probably not set a default value for the parameter, though.

Reply:
updated @shenglol